### PR TITLE
feat(Sprockets) add sprockets 2 and 3 to Appraisals

### PR DESCRIPTION
**Pull request by rmosolgo**
There are some known bugs with Sprockets incompatibilities and I'd like to make those more reproducible (and maintain compatibility with Sprockets 2)

### CODE REVIEW

**Comment by rmosolgo** [comment]
image

**Comment by vipulnsward** [comment]
🚀

File: lib/react/rails/engine.rb [hunk 1]
```diff
@@ -2,12 +2,11 @@ module React
   module Rails
     class Engine < ::Rails::Engine
       initializer "react_rails.setup_engine", :group => :all do |app|
-        sprockets_env = app.assets || Sprockets # Sprockets 3.x expects this in a different place
         if Gem::Version.new(Sprockets::VERSION) >= Gem::Version.new("3.0.0")
-          sprockets_env.register_mime_type("application/jsx", extensions: [".jsx", ".js.jsx", ".es.jsx", ".es6.jsx"])
-          sprockets_env.register_transformer("application/jsx", "application/javascript", React::JSX::Processor)
+          Sprockets.register_mime_type("application/jsx", extensions: [".jsx", ".js.jsx", ".es.jsx", ".es6.jsx"])
+          Sprockets.register_transformer("application/jsx", "application/javascript", React::JSX::Processor)
```
Comment (vipulnsward) [inline lib/react/rails/engine.rb]: Is this taken care of by sprockets?

Comment (rmosolgo) [inline lib/react/rails/engine.rb]: Maybe not -- I was "refactoring" but I misread that line above it!

Comment (rmosolgo) [inline lib/react/rails/engine.rb]: Lol, obviously that broke stuff like crazy, oops :grimacing:

**Comment by rmosolgo** [comment]
Ok, the remaining failure looks like the match of #415 

(Rails 4.0 Gemfile is using Sprockets 3 https://travis-ci.org/reactjs/react-rails/jobs/104466644#L345)

**Comment by rmosolgo** [comment]
I removed that failing test. It's important to figure that out, but it doesn't have to block this improvement to the test suite.

**Comment by rmosolgo** [comment]
I removed the tests for 1.9 and 2.0. Apparently 1.9 is EOL and 2.0 is soon to go! (And we have _so many_ test runs, it's just too much!)

**Comment by rmosolgo** [comment]
Wow, i don't know why `click_button` was so much more reliable than `click_link` in that case but ... :relieved:

**Pull request merged**